src/game/game.js

```diff
@@ -21,33 +28,73 @@ class Game {
         handle: () => this.world.getPlayer().moveRight(),
       },
     };
-
-    this.world = new World();
-    this.render = new Render(canvas, this.world, this.inputController);
   }
 
   createGame() {
+    this.world = new World();
+    this.render = new Render(
+      this.canvas,
+      this.world,
+      this.inputController,
+      this.controllerCallback,
+    );
     this.world.generate();
     this.render.animate();
   }
 
+  destroyGame() {
+    this.render = undefined;
+    this.world = undefined;
+    this.removeEventListeners();
+  }
+
   startGame() {
     this.addEventListeners();
   }
 
+  pauseGame() {
+    this.removeEventListeners();
+    this.render.pause();
+  }
+
+  resumeGame() {
+    this.render.resume();
+    this.addEventListeners();
+  }
+
+  restartGame() {
+    this.removeEventListeners();
+    this.world.removeObjects();
+    this.world.generate();
+    this.addEventListeners();
+    this.render.restart();
+  }
+
+  handleInputKeyDown(event) {
+    if (event.key === 'Escape') {
+      this.controllerCallback('pause');
+    }
+    if (this.inputController[event.key]) {
+      this.inputController[event.key].pressed = true;
+    }
+  }
+
+  handleInputKeyUp(event) {
+    if (this.inputController[event.key]) {
+      this.inputController[event.key].pressed = false;
+    }
+  }
+
   // Register user input listeners
   addEventListeners() {
-    window.addEventListener('keydown', (event) => {
-      if (this.inputController[event.key]) {
-        this.inputController[event.key].pressed = true;
-      }
-    });
-
-    window.addEventListener('keyup', (event) => {
-      if (this.inputController[event.key]) {
-        this.inputController[event.key].pressed = false;
-      }
-    });
+    window.addEventListener('keydown', this.handleInputKeyDown);
+    window.addEventListener('keyup', this.handleInputKeyUp);
+  }
+
+  // Remove user input for game
+  removeEventListeners() {
+    window.removeEventListener('keydown', this.handleInputKeyDown);
+    window.removeEventListener('keyup', this.handleInputKeyUp);
```

Comment:
This is just formality, but it is a good practice when writing a "destructor" to destruct things in opposite order of their creation. Since you added the 'keyup' listener last, it would be nice to remove it first. This does not change the functionality of the program, but this is a good practice and helps to dodge nasty software dependency issues.